src/utils/dom.ts

```diff
@@ -0,0 +1,15 @@
+export function isTouchDevice(): boolean {
+    if (typeof window === 'undefined') {
+        return false;
+    }
+
+    if (typeof navigator !== 'undefined' && navigator.maxTouchPoints > 0) {
+        return true;
+    }
+
+    if (window.matchMedia && window.matchMedia('(any-pointer:coarse)').matches) {
+        return true;
+    }
+
+    return Boolean(window.TouchEvent || 'ontouchstart' in window);
```

Comment:
It looks like this check will be true for non-touch devices too, right?

Reply:
Seem's like browser provides `TouchEvent` API also [for devices with trackpad](https://developer.mozilla.org/en-US/docs/Web/API/TouchEvent#:~:text=touch%20screen%20or%20trackpad), so you right! I deleted `window.TouchEvent` from this condition